### PR TITLE
vhost_user: expand error messages for clarity

### PR DESCRIPTION
**Pull request by stsquad**
### Summary of the PR

The InvalidOperation error type covers a wide range of error cases
which can be inscrutable when passed to the user. To help with this
we:

  - add a textual reason field to the error message
  - create InactiveFeature for use of un-neogitated features
  - create InactiveOperation for use of un-negotiated protocol extensions

### Requirements

Before submitting your PR, please make sure you addressed the following
requirements:

- [x] All commits in this PR are signed (with `git commit -s`), and the commit
  message has max 60 characters for the summary and max 75 characters for each
  description line.
- [x] All added/changed functionality has a corresponding unit/integration
  test.
- [x] Any newly added `unsafe` code is properly documented.

### CODE REVIEW

**Comment by slp** [comment]
LGTM, I think we can merge this one once the coverage number is fixed to the new value. Thanks @stsquad !

**Comment by stsquad** [comment]
I have a small patches waiting for vhost-user-master and vhost-device once the version for this crate is bumped .

**Review by stefano-garzarella — APPROVED** [review]

**Review by sboeuf — APPROVED** [review]

**Comment by stsquad** [comment]
Does this PR not get automatically merged once approved or does it need to be done manually?

**Comment by sboeuf** [comment]
> Does this PR not get automatically merged once approved or does it need to be done manually?

No it still has to be merged manually.

**Pull request merged**
